### PR TITLE
Add Pinterest associated websites

### PR DESCRIPTION
**Pull request by MartinDevi**
### Overall Checklist
- [x] I agree to the project's [Developer Certificate of Origin](https://github.com/apple/password-manager-resources/blob/main/DEVELOPER_CERTIFICATE_OF_ORIGIN.md)
- [x] The top-level JSON objects are sorted alphabetically
- [x] There are no [open pull requests](https://github.com/apple/password-manager-resources/pulls) for the same update

#### for websites-with-shared-credential-backends.json
- [ ] There's evidence the domains are related (SSL certificates, DNS entries, valid links between sites, legal documents etc.)
- [x] The new group serves login pages on each of the included domains, and those login page accept accounts from the others. (For example, we don't associate `google.co.il` to `google.com`, because `google.co.il` redirects to `accounts.google.com` for authentication.)

### CODE REVIEW

**Comment by rmondello** [comment]
These seem good, except pintrest.com, which is a redirect.

**Review by rmondello — CHANGES_REQUESTED** [review]
Should we remove pintrest.com?

**Comment by Cldfire** [comment]
The `pintrest.com` entry would fit very well into the format defined in https://github.com/apple/password-manager-resources/issues/389 😄

**Comment by Cldfire** [comment]
Thinking about this some more, it might not be worth adding `pintrest.com` to the format proposed in #389.

We can assume that `pintrest.com` was registered by Pinterest to prevent someone else from buying up the (very similar) domain and using it for phishing. There's no reason why a user of a password manager should encounter this redirect during normal web browsing. In fact, should the user end up with credentials associated with the `pintrest.com` domain, I would argue that that's an issue that should be corrected by the user; a password manager shouldn't attempt to transparently adjust for that.

With that in mind, I would agree with @rmondello; `pintrest.com` should be removed here.

**Comment by MartinDevi** [comment]
Removed `pintrest.com`

File: quirks/websites-with-shared-credential-backends.json [hunk 1]
```diff
@@ -288,6 +288,30 @@
...
         "optum.com"
     ],
+    [
+        "pinterest.com",
+        "pinterest.ca",
+        "pinterest.co.uk",
+        "pinterest.fr",
+        "pinterest.de",
+        "pinterest.es",
+        "pin.it",
```
Comment (Cldfire) [inline quirks/websites-with-shared-credential-backends.json:298]: I'd say similar logic applies here; `pin.it` is a redirect and should be removed from this PR.

**Comment by Cldfire** [comment]
Everything LGTM after removing `pin.it` :)

**Review by Cldfire — APPROVED** [review]

**Review by rmondello — APPROVED** [review]

**Pull request merged**